src/components/TicketsTable.tsx

```diff
@@ -0,0 +1,80 @@
+import { DataTable } from 'primereact/datatable';
+import { Column } from 'primereact/column';
+import Link from 'next/link';
+import type { ServiceRequestSummary } from '@types';
+
+export interface TicketsTableProps {
+  items: ServiceRequestSummary[]
+}
+
+function OwnerAndPetTemplate({ client, pet, id }) {
+  return (
+    <div key={id}>
+      <Link href={`?ticket=${id}`}>
+        <div className="font-bold">{pet}</div>
+        <div className="capitalize">{client}</div>
+      </Link>
+
+    </div>
+  );
+}
+
+function CreatedAtTemplate({ created_at, id }) {
+  return (
+    <span key={id}>
+      <Link href={`?ticket=${id}`}>
+        {new Date(created_at).toLocaleDateString('en-US', {
+          day: '2-digit', month: '2-digit', year: 'numeric',
+        })}
+      </Link>
+    </span>
+  );
+}
+
+function descriptionView({ id, description }) {
+  return (
+    <Link href={`?ticket=${id}`} key={id}>
+      {description}
+    </Link>
+  );
+}
+function catergoryView({ id, category }) {
+  return (
+    <Link href={`?ticket=${id}`} key={id}>
+      {category}
+    </Link>
+  );
+}
+function TeamMemberView({ id, team_member }) {
+  return (
+    <Link className="text-red-500" href={`?ticket=${id}`} key={id}>
+      {team_member}
+    </Link>
+  );
+}
+
+function TicketsTable({ items }: TicketsTableProps, i) {
+  console.log('items', items);
+  return (
+    <DataTable
+      value={items}
+      paginatorTemplate="FirstPageLink PrevPageLink PageLinks NextPageLink LastPageLink CurrentPageReport RowsPerPageDropdown"
+      dataKey="id"
+      paginator
+      emptyMessage="No data found."
+      className="datatable-responsive"
+      currentPageReportTemplate="Showing {first} to {last} of {totalRecords} tickets"
```

Comment:
Where does total records come from?

Reply:
it looks like that's just how primereacts data table works. there is a preset list of keywords that can be used with <mark>'currentPageReportTemplate'</mark> prop


It's not documented well in the [datatables docs page](https://primereact.org/datatable/), but dug in the code for [[1].paginator](https://github.com/primefaces/primereact/blob/8625e49922a94eed764f80d6cfd486653fad4b61/components/lib/paginator/Paginator.js#L235C29-L235C77) and [[2].currentPageReport prop](https://github.com/primefaces/primereact/blob/8625e49922a94eed764f80d6cfd486653fad4b61/components/lib/paginator/CurrentPageReport.js#L12C5-L27C57) and they calculate those numbers based on the the items that are passed into <mark>'value'</mark> prop. here's how <mark>'datatable'</mark> component calculates and passes all the numbers to the paginator component [link](https://github.com/primefaces/primereact/blob/8625e49922a94eed764f80d6cfd486653fad4b61/components/lib/datatable/DataTable.js#L1912C13-L1929C15)

Reply:
It is part of the primreact design for its table

Reply:
> Where does total records come from?

Like Adele said it is part of this components library, there is a set of props we can utilize to get certain features and @AdelBeit  agreed is not documented well.

Reply:
Ok gotcha, so if we only query 10 records, then the totalRecords would show `10` even if there was more in the database?

Reply:
> Ok gotcha, so if we only query 10 records, then the totalRecords would show `10` even if there was more in the database?

totalRecords was tied to how big the array of objects was being passed in this case since we had the original limit query to 10 from db it would be 10.  But `totalRecords` will just put any number based on the sized of an array we pass into it if an array has length of 1000, `totalRecords` = 1000 the thing that was limiting this was the .limit(10) from the query which I did not get at first